testing/upgradetest/watcher.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package upgradetest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/utils"
+)
+
+// FastWatcherCfg is configuration that makes the watcher run faster.
+const FastWatcherCfg = `
+agent.upgradetest.watcher:
+  grace_period: 1m
+  error_check.interval: 15s
+  crash_check.interval: 15s
+`
+
+// ConfigureFastWatcher writes an Elastic Agent configuration that
+// adjusts the watcher to be faster.
+func ConfigureFastWatcher(ctx context.Context, f *atesting.Fixture) error {
+	return f.Configure(ctx, []byte(FastWatcherCfg))
```

Comment:
Does this work if the agent is Fleet managed?

Reply:
I think from previous testing it doesn't, since the only config the agent respects is the one from Fleet. We should note this here to avoid people using this where it doesn't help (unless I'm wrong about it).

Reply:
I think your correct, I will add a note.